cmd/gitserver/server/vcs_syncer.go

```diff
@@ -302,3 +307,106 @@ func (s *PerforceDepotSyncer) RemoteShowCommand(ctx context.Context, remoteURL *
 	// Remote info is encoded as in the current repository
 	return exec.CommandContext(ctx, "git", "remote", "show", "./"), nil
 }
+
+type MavenArtifactSyncer struct {
+	Config *schema.MavenConnection
+}
+
+var _ VCSSyncer = &MavenArtifactSyncer{}
+
+func (s MavenArtifactSyncer) Type() string {
+	return "maven"
+}
+
+// IsCloneable checks to see if the VCS remote URL is cloneable. Any non-nil
+// error indicates there is a problem.
+func (s MavenArtifactSyncer) IsCloneable(ctx context.Context, remoteURL *vcs.URL) error {
+	groupID, artifactID, version := reposource.DecomposeMavenPath(remoteURL.Path)
+	exists, err := coursier.Exists(ctx, s.Config, groupID, artifactID, version)
+	if err != nil {
+		return err
+	}
+	if exists {
+		return nil
+	}
+
+	return errors.New(fmt.Sprintf("Maven repo %v not found", remoteURL))
```

Comment:
```suggestion
	return errors.Errorf("Maven repo %q not found", remoteURL)
```

This is the shortcut 😁

PS, `%q` quotes the string, nit though.